napari/layers/tracks/_track_utils.py

```diff
@@ -66,31 +71,19 @@ class TrackManager:
         Track ID for each vertex in track_vertices.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, data: np.ndarray) -> None:
         # store the raw data here
-        self._data = None
+        self.data = data
         self._feature_table = _FeatureTable()
-        self._order = None
-
-        # use a kdtree to help with fast lookup of the nearest track
-        self._kdtree = None
-
-        # NOTE(arl): _tracks and _connex store raw data for vispy
-        self._points = None
         self._points_id = None
-        self._points_lookup = None
-        self._ordered_points_idx = None
```

Comment:
Do you want to move some of those to typed class attributes ?

And for next reviewer, those are indeed all set in the data setter, included _id2idxs lower in this `__init__`

Reply:
By typed class attributes, do you mean a class attribute that just has a type annotation and no default value? If so I think that would be a reasonable thing to do in this PR.

Reply:
I think for this case, we should declare those as typed class attributes without defaults in the class itself (i.e. above `__init__`) as I think @Carreau is suggesting. That way, we clarify all the attributes of the class, don't need them to be optional, and also make the initializer simple and consistent with the data setter.